src/shogun/mathematics/linalg/BaseVector.h

```diff
@@ -54,7 +54,7 @@ struct BaseVector
 	}
 
 	/** Data Storage */
-	virtual bool onGPU() = 0 ;
+	virtual bool onGPU() const = 0 ;
```

Comment:
Okay thanks. But maybe we won't have to do the casting, if we can do what you discussed earlier in the gist @OXPHOS made - have it in the same class and call onGPU method. I can't seem to remember what was stopping us from doing that.